applications/MappingApplication/custom_python/add_custom_mappers_to_python.cpp

```diff
@@ -0,0 +1,191 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Philipp Bucher, Jordi Cotela
+//
+// See Master-Thesis P.Bucher
+// "Development and Implementation of a Parallel
+//  Framework for Non-Matching Grid Mapping"
+
+// System includes
+
+// External includes
+#include <boost/python.hpp>
+
+// Project includes
+#include "custom_utilities/mapper_flags.h"
+#include "custom_python/add_custom_mappers_to_python.h"
+#include "custom_utilities/mapper_factory.h"
+
+// Mapper base class
+#include "custom_mappers/mapper.h"
+
+// Matrix-free Mappers
+#include "custom_mappers/nearest_neighbor_mapper.h"
+#include "custom_mappers/nearest_element_mapper.h"
+
+
+namespace Kratos
+{
+
+namespace Python
+{
+
+// Wrapper functions for taking a default argument for the flags // TODO inline? Jordi
+inline void UpdateInterface(Mapper& dummy)
```

Comment:
The `inline` is fine (the compiler treats it only as a suggestion, anyway), but I'm not sure all those "dummy" functions are still needed: you are using the function pointers defined later to link to python.
If they are no longer required, please remove them.

Reply:
I need those functions to have the functions in Python exposed with the default arguments
e.g. the `UpdateInterface` Function I can call with 
~~~py
UpdateInterface()
UpdateInterface(Kratos::Flags)
UpdateInterface(double Search Radius)
UpdateInterface(Kratos::Flags, double Search Radius)
~~~
Same for the mapping functions

If you know an alternative, I am happy if I can achieve this behavior in an easier way

Reply:
I've never used, but boost::python gives you support to do this automatically with `BOOST_PYTHON_MEMBER_FUNCTION_OVERLOADS`(see http://www.boost.org/doc/libs/1_65_1/libs/python/doc/html/tutorial/tutorial/functions.html#tutorial.functions.default_arguments.boost_python_member_function_ove).
If you want to keep doing it the way it is done now, it is OK for me too.

Reply:
I can try to do it but I am not sure if it is worth the effort bcs I saw that @RiccardoRossi is starting to use pybind and when Kratos switches to pybind I could do the change properly
@RiccardoRossi what is the status of using pybind?

Reply:
@philbucher the pybind change won't be immediate, but I see your point. I hope it will have equivalent functionality however.
In any case, feel free to merge as is, it is OK for me.

Reply:
Hi @jcotela I tried the Boost Macro but it seems like it is not possible to use it. It seems to me like the macro creates some static object, which is not allowed in a local scope (i.e. `AddCustomMappersToPython`) in c++
~~~sh
/home/philippb/software/Kratos_aux/applications/MappingApplication/custom_python/add_custom_mappers_to_python.cpp:149:5: error: local class ‘struct Kratos::Python::AddCustomMappersToPython()::update_interface_overloads::non_void_return_type’ shall not have static data member ‘const int Kratos::Python::AddCustomMappersToPython()::update_interface_overloads::non_void_return_type::n_funcs’ [-fpermissive]
     BOOST_PYTHON_MEMBER_FUNCTION_OVERLOADS(update_interface_overloads, UpdateInterface, 0, 2);
~~~
I did a search, it does not seem to be used anywhere else in Kratos.
Have you used this somwhere else?

Reply:
No, I have never used it (I tend to not provide default arguments). It is just that I saw it in the boost docs as the "recommended" usage. If it doesn't work, just forget about it.

Reply:
Lets see maybe I find another way to do it better in the future

Then I am waiting for @pooyan-dadvand s approval